cfme/tests/configure/test_rest_access_control.py

```diff
@@ -819,3 +820,103 @@ def test_delete_users_from_collection(self, users):
             initialEstimate: 1/4h
         """
         delete_resources_from_collection(users)
+
+
+COMMON_FEATURES = [
+    "Services",
+    "Compute",
+    "Configuration",
+    "Networks",
+    "Storage",
+    "Control",
+    "Automation",
+    "Monitor",
+    "Help",
+    "API",
+    "Service UI",
+]
+
+FEATURES_511 = COMMON_FEATURES + [
+    "Overview",
+    "Migration",
+    "User Settings",
+    "All VM and Instance Access Rules",
+    "Main Configuration",
+]
+
+FEATURES_510 = COMMON_FEATURES + [
+    "Cloud Intel",
+    "Red Hat Insights",
+    "Optimize",
+    "Access Rules for all Virtual Machines",
+]
+
+
+@pytest.fixture(params=[True, False], ids=["individual", "everything"])
+def create_role(appliance, request):
+    role_name = "API-role-{}".format(fauxfactory.gen_alpha(4))
+    features = FEATURES_511 if appliance.version > "5.11" else FEATURES_510
+
+    view = navigate_to(appliance.collections.roles, "Add")
+    view.name_txt.fill(role_name)
+
+    view.features_tree.check_node(*["Everything"])
+    if request.param:
+        view.features_tree.uncheck_node(*["Everything"])
+        for feature in features:
+            view.features_tree.check_node(*["Everything", feature])
+
+    view.add_button.click()
+    view.flash.assert_no_error()
+
+    role = appliance.rest_api.collections.roles.get(name=role_name)
+    yield role
+    role.action.delete
+
+
+@pytest.fixture
+def role_api(appliance, request, create_role):
+    group = _groups(request, appliance, create_role)
+    user, user_data = _users(request, appliance, group=group.description)
+
+    yield appliance.new_rest_api_instance(
+        entry_point=appliance.rest_api._entry_point,
+        auth=(user[0].userid, user_data[0]["password"]),
+    )
+    user[0].action.delete
```

Comment:
Q: Why `user[0]`?

Reply:
Because `gen_data.users` creates users and returns a list. And I need to access the user, so I use `user[0]`.

Reply:
On second thought, I don't really need to delete the user, it will be automatically deleted by `create_skeleton` from gen_data.